langchain/vectorstores/pinecone.py

```diff
@@ -0,0 +1,162 @@
+        try:
+            import pinecone
+        except ImportError:
+            raise ValueError(
+                "Could not import pinecone python package. "
+                "Please it install it with `pip install pinecone-client`."
+            )
+
+        # Embed and create the documents
+        docs = []
+        ids = []
+        for i, text in enumerate(texts):
+            id = str(uuid.uuid4())
+            embedding = self._embedding_function(text)
+            metadata = metadatas[i] if metadatas else {}
+            metadata[self._text_key] = text
+            docs.append((id, embedding, metadata))
+            ids.append(id)
+        # upsert to Pinecone
+        self._index.upsert(vectors=docs)
+        return ids
+
+    def similarity_search(self, query: str, k: int = 4) -> List[Document]:
+        """Look up similar documents in pinecone."""
+        query_obj = self._embedding_function(query)
+        docs = []
+        results = self._index.query([query_obj], top_k=5, include_metadata=True)
+        for res in results["matches"]:
+            metadata = res["metadata"]
+            text = metadata.pop(self._text_key)
+            docs.append(Document(page_content=text, metadata=metadata))
+        return docs
+
+    @classmethod
+    def from_texts(
+        cls,
+        texts: List[str],
+        embedding: Embeddings,
+        metadatas: Optional[List[dict]] = None,
+        batch_size: int = 32,
+        text_key: str = "text",
+        **kwargs: Any,
+    ) -> Pinecone:
+        """Construct Pinecone wrapper from raw documents.
+
+        This is a user friendly interface that:
+            1. Embeds documents.
+            2. Creates a new index for the embeddings in Pinecone
+            3. Adds the documents to the newly created Pinecone index
+
+        This is intended to be a quick way to get started.
+
+        Example:
+            .. code-block:: python
+
+                from langchain import Pinecone
+                from langchain.embeddings import OpenAIEmbeddings
+                embeddings = OpenAIEmbeddings()
+                pinecone = Pinecone.from_texts(
+                    texts,
+                    embeddings
+                    api_key="PINECONE_API_KEY"
+                )
+        """
+        api_key = get_from_dict_or_env(kwargs, "api_key", "PINECONE_API_KEY")
+        try:
+            import pinecone
+        except ImportError:
+            raise ValueError(
+                "Could not import pinecone python package. "
+                "Please it install it with `pip install pinecone-client`."
+            )
+        try:
+            pinecone.init(api_key=api_key, environment="us-west1-gcp")
+        except ValueError as e:
+            raise ValueError("Pinecone initialization failed. " f"Got error: {e} ")
+
+        # Create first embedding to get correct dims
+        res = embedding.embed_query(texts[0])
+        index_name = uuid.uuid4().hex
```

Comment:
given the pinecone free API only supports one index at a time, is this annoying? To rerun this class you'd have to use since the index name is random..
`index = pinecone.list_indexes()[0]
pinecone.delete_index(index)
`

Reply:
can we just assume that this is done outside? and that `index_name` is passed in? would make a lot of this stuff (inlcuding having to do the pinecone.init) easier

Reply:
yeah, that might be best. was trying to make usage easier / more aligned with the other vectorstores, but given the pinecone setup that might be best